interop-testing/src/main/java/io/grpc/testing/integration/GrpclbFallbackTestClient.java

```diff
@@ -0,0 +1,296 @@
+          + "LB/backend addresses becoming unroutable"
+          + "\n      slow_fallback_before_startup : fallback before LB connection "
+          + "due to LB/backend addresses being blackholed"
+          + "\n      slow_fallback_after_startup : fallback after startup due to "
+          + "LB/backend addresses becoming blackholed"
+          + "\n      Default: " + c.testCase
+      );
+      System.exit(1);
+    }
+  }
+
+  private ManagedChannel createChannel() {
+    if (!customCredentialsType.equals("compute_engine_channel_creds")) {
+      throw new AssertionError(
+          "This test currently only supports "
+          + "--custom_credentials_type=compute_engine_channel_creds. "
+          + "TODO: add support for other types.");
+    }
+    ComputeEngineChannelBuilder builder = ComputeEngineChannelBuilder.forTarget(serverUri);
+    builder.keepAliveTime(3600, TimeUnit.SECONDS);
+    builder.keepAliveTimeout(20, TimeUnit.SECONDS);
+    return builder.build();
+  }
+
+  void setUp() {
+    channel = createChannel();
+    blockingStub = TestServiceGrpc.newBlockingStub(channel);
+  }
+
+  private void tearDown() {
+    try {
+      channel.shutdownNow();
+      channel.awaitTermination(1, TimeUnit.SECONDS);
+    } catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
+  private void runShellCmd(String cmd) throws Exception {
+    logger.info("Run shell command: " + cmd);
+    ProcessBuilder pb = new ProcessBuilder("bash", "-c", cmd);
+    pb.redirectErrorStream(true);
+    Process process = pb.start();
+    logger.info("Shell command merged stdout and stderr: "
+        + CharStreams.toString(
+            new InputStreamReader(process.getInputStream(), UTF_8)));
+    int exitCode = process.waitFor();
+    logger.info("Shell rommand exit code: " + exitCode);
+    assertEquals(0, exitCode);
+  }
+
+  private enum RpcMode {
+    FailFast,
+    WaitForReady,
+  }
+
+  private GrpclbRouteType doRpcAndGetPath(
+      TestServiceGrpc.TestServiceBlockingStub blockingStub,
+      int deadlineSeconds,
+      RpcMode rpcMode) {
+    logger.info("doRpcAndGetPath deadlineSeconds: " + deadlineSeconds + " rpcMode: "
+        + rpcMode);
+    final SimpleRequest request = SimpleRequest.newBuilder()
+        .setFillGrpclbRouteType(true)
+        .build();
+    GrpclbRouteType result = GrpclbRouteType.GRPCLB_ROUTE_TYPE_UNKNOWN;
+    try {
+      SimpleResponse response = null;
+      if (rpcMode == RpcMode.WaitForReady) {
+        response = blockingStub
+          .withDeadlineAfter(deadlineSeconds, TimeUnit.SECONDS)
+          .withWaitForReady()
+          .unaryCall(request);
+      } else {
+        response = blockingStub
+          .withDeadlineAfter(deadlineSeconds, TimeUnit.SECONDS)
+          .unaryCall(request);
+      }
+      result = response.getGrpclbRouteType();
+    } catch (StatusRuntimeException ex) {
+      logger.warning("doRpcAndGetPath failed. Status: " + ex);
```

Comment:
I'm surprised this exception isn't propagated. It seems if this triggers `result` will be `UNKNOWN` which will fail in the assertion below and would be a misleading message.

Reply:
This was broken, sorry. Changed to return `UNKNOWN` early in case of RPC failure